dsfhub/resource_common.go

```diff
@@ -519,14 +625,25 @@ func connectGateway(ctx context.Context, m interface{}, assetId string, resource
 		return err
 	}
 
-	err2 := waitUntilAuditState(ctx, true, resourceType, assetId, m)
-	if err2 != nil {
-		return err2
+	// ensure asset is synced to gateway
+	err = waitForRemoteSyncState(ctx, resourceType, assetId, m)
+	if err != nil {
+		return err
+	}
+
+	// confirm asset is connected to gateway
+	isAuditPullEnabled, err := checkAuditState(ctx, m, assetId, resourceType)
+	if err != nil {
```

Comment:
can combine this and the following if statement into one with an or `||` (here and in `disconnectGateway`)

Reply:
I disagree, but mostly because now that I'm looking at this a second time, if we even get to this point
```
if !isAuditPullEnabled {
```
it means `err == nil`. I'll add a better error message in the following check.